internal/storage/storage_handle.go

```diff
@@ -44,13 +45,14 @@ type StorageHandle interface {
 }
 
 type storageClient struct {
-	client *storage.Client
+	client               *storage.Client
+	storageControlClient *control.StorageControlClient
 }
 
 // Followed https://pkg.go.dev/cloud.google.com/go/storage#hdr-Experimental_gRPC_API to create the gRPC client.
-func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.StorageClientConfig) (sc *storage.Client, err error) {
+func createGRPCClientHandle(ctx context.Context, clientConfig *storageutil.StorageClientConfig) (sc *storage.Client, controlClient *control.StorageControlClient, err error) {
```

Comment:
instead of modifying existing method, we should create a new method which will do same thing i.e creating a grpc client but the client will be control client and it should read from a different flag which should be folder api client protocol